src/pages/Error404.jsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+
+export const Error404 = () => {
+   const [isLoaded, setIsLoaded] = useState(false)
+
+   useEffect(() => {
```

Comment:
I know we used this solution before for the firebase state, but for handling the assets do not use the react life cycle, instead look for solutions in the vite documentation.

opposite to the firebase case this is something that we can handle on our own.

keeping components stateless is a way to optimize the performance of a react app.